constellations/gemini-admin/src/account/accountTransactionEdit.js

```diff
@@ -0,0 +1,245 @@
+                .then(() => successCallback())
+                .catch((err) => failCallback(err));
+        }
+    };
+
+    onModalOkSaved = () => {
+        this.onModalDismiss();
+        window.location.hash = "accounts";
+    };
+
+    onModalDismiss = () => {
+        this.setState({
+            showDeleteModal: false,
+            showSaveModal: false,
+        });
+    };
+
+    render = () => {
+        const isEdit = this.state.isEdit;
+        var title = isEdit ? "Edit Transaction" : "Add Transaction";
+        title += " (Account No. " + this.state.accountId + ")";
+
+        const modalDiv = renderModal(
+            this.state.showSaveModal,
+            this.state.showDeleteModal,
+            this.onModalOkSaved,
+            this.onModalDeleteConfirm,
+            this.onModalDismiss
+        );
+
+        const typeOptions = this.state.types.map((type, index) => {
+            return (
+                <option value={type} key={index}>
+                    {type}
+                </option>
+            );
+        });
+
+        let deleteButton = <div></div>;
+        if (isEdit) {
+            deleteButton = (
+                <button className="btn-delete" onClick={this.onClickDelete}>
+                    Delete
+                </button>
+            );
+        }
+
+        return (
+            <div id="view-transaction-edit">
+                {modalDiv}
+                <h1>{title}</h1>
+
+                <div id="transaction-type">
+                    <h2>Type</h2>
+                    <h4>Enter the type of transaction</h4>
+                    <select
+                        className="dropdown"
+                        value={this.state.type}
+                        onChange={(e) => this.handleChange(e, "type")}>
+                        {typeOptions}
+                    </select>
+                </div>
+
+                <InputText
+                    label="Amount"
+                    value={this.state.amount}
+                    onChangeCallback={(e) => this.handleChange(e, "amount")}
+                    required={true}
+                    description="Enter the amount of money"
+                    validators={[
+                        {
+                            validate: (amount) =>
+                                !isNaN(amount) && amount != "",
+                            message: "You must input an amount",
+                        },
+                        {
+                            validate: (amount) =>
+                                !(this.state.type != "charge" && amount < 0),
+                            message: "Payment must be positive",
+                        },
+                        {
+                            validate: (amount) =>
+                                !(this.state.type == "charge" && amount > 0),
+                            message: "Charge must be negative",
+                        },
+                    ]}
```

Comment:
it's a long validator, maybe consider putting it into a variable so this part becomes `validators={validatorFns}`